src/main/java/Stations.java

```diff
@@ -4,16 +4,98 @@
 import java.io.Reader;
 import java.nio.file.Files;
 import java.nio.file.Paths;
-import java.util.ArrayList;
-import java.util.List;
+import java.util.*;
 
 public class Stations {
 
     List<Station> features;
+    
+
+    public Path findShortestPath(Station startingStation, Station endingStation) throws IOException
+    {
+        Path shortestPath = new Path(Integer.MAX_VALUE);
+        startingStation.shortestPath = new Path(startingStation);
+
+        // create priority queue
+        PriorityQueue<Station> stationsToCheck = new PriorityQueue<Station>(new PathComparator());
```

Comment:
`Mark all nodes unvisited. Create a set of all the unvisited nodes called the unvisited set.`

This is the first instruction from the wikipedia page.

Is that what your code does?